versioned_docs/version-5.x/configuring-links.md

```diff
@@ -0,0 +1,355 @@
+
+What if we wanted a specific screen to be always present in the state object for a navigator? For example, if we had a URL that would open `Home` screen, but we would like to be able to navigate to `Profile` from it by using navigation's `navigation.goBack()` method. It is possible by defining `initialRouteName` for a navigator. It would look like this then:
+
+```js
+config = {
+  HomeStack: {
+    path: 'stack',
+    initialRouteName: 'Profile',
+    screens: {
+      Home: 'home',
+      Profile: 'user',
+    },
+  },
+  Settings: 'settings',
+};
+```
+
+### Passing parameters to screens
+
+A common case in having navigation in the application is passing parameters to the screens while navigating between them. In our example, we want the `Profile` screen to have the `id` param. While in app, we pass that param as a property of an object in the second argument of `navigation.navigate()` function. If we want to pass such param while opening the screen through deep linking, we can achieve it by extending the config for the screen. For `id` in `Profile`, it would be `Profile: 'user/:id'`. `/` is the separator between the path and each of the parameters. Passing URL `user/Wojciech` would resolve to `Profile` screen with a 'Wojciech' as a value of the `id` param in the screen params. It can be extracted from `route.params.id` in the screen component.
+
+Sometimes we want to parse the param in a specific way. By default, all params are parsed to the screen as strings, but we can define a `parse` property, which will handle parsing, and `stringify` property to convert it back to a string. If we wanted to resolve `user/Wojciech/22` to user's age as a number and a modified id, we could make `Profile`'s config look like this:
+
+```js
+Profile: {
+  path: 'user/:id/:age',
+  parse: {
+    id: id => `user-${id}`,
+    age: Number,
+  },
+  stringify: {
+    id: id => id.replace(/^xxx-/)
+  }
+}
+```
+
+Similarly to web URLs, there is also an option of using query params, which can only be applied to the deepest screen in the URL. To apply it to our config, the URL would look like this: `user?id=Wojciech&age=22` and config would not include params in the `path` property:
+
+```js
+Profile: {
+  path: 'user',
+  parse: {
+    id: id => `user-${id}`,
+    age: Number,
+  },
+  stringify: {
+    id: id => id.replace(/^xxx-/)
+  }
+}
+```
+
+### Specific configuration
+
+Here we will mention some of the consequences of using nested navigators in the config.
+
+1. Using `initialRouteName` makes the screen always appear in the state of a navigator. It makes it impossible to pass params to that screen through the URL unless the screen is explicitly mentioned in it. It is then reasonable for that screen not to take any params or to provide default ones.
```

Comment:
for each of these points it would be good to summarize them with a bold title beforehand, eg:

```suggestion
1. **Using `initialRouteName` makes the screen always appear in the state of a navigator.** It makes it impossible to pass params to that screen through the URL unless the screen is explicitly mentioned in it. It is then reasonable for that screen not to take any params or to provide default ones.
```